### PR TITLE
feat(firestore-translate): text multiple fields update

### PR DESCRIPTION
**Pull request by dackers86**
### Testing

Single
image

Multiple

image

### CODE REVIEW

**Review by russellwheatley — CHANGES_REQUESTED** [review]
I think this [function](https://github.com/firebase/extensions/blob/961d248fcc15fc6511d8e57f7116b199d30b1a07/firestore-translate-text/functions/src/index.ts#L119) needs to be updated to reflect there could be mutliple fields. For instance, it is checking `const inputHasChanged = inputAfter !== inputBefore;` when they could be two object references, and thus never the same value.

File: firestore-translate-text/functions/src/index.ts [hunk 1]
```diff
@@ -173,13 +194,58 @@ const translateDocument = async (
...
+    if (fn) await fn();
+  }
+
+  return updateTranslations(snapshot, translations);
+};
+
+const translateDocument = async (
+  snapshot: admin.firestore.DocumentSnapshot
+): Promise<void> => {
+  const input: string = extractInput(snapshot);
```
Comment (Ehesp) [inline firestore-translate-text/functions/src/index.ts:239]: Not specifically a `string` now.

File: firestore-translate-text/functions/src/index.ts [hunk 2]
```diff
@@ -173,13 +194,58 @@ const translateDocument = async (
...
+
+  if (typeof input === "object") {
+    await translateMultiple(input, snapshot);
+    return Promise.resolve();
```
Comment (Ehesp) [inline firestore-translate-text/functions/src/index.ts:243]: Can just `return` here, since it's already a promise, rather than creating a new instance.

**Review by Ehesp — CHANGES_REQUESTED** [review]

File: firestore-translate-text/functions/src/index.ts [hunk 3]
```diff
@@ -118,36 +118,56 @@ const handleUpdateDocument = async (
...
+  if (typeof inputBefore !== typeof inputAfter) {
     logs.documentUpdatedChangedInput();
     await translateDocument(after);
-  } else if (inputBefore) {
-    logs.documentUpdatedDeletedInput();
-    await updateTranslations(after, admin.firestore.FieldValue.delete());
+    return;
+  }
+
+  let beforeValues = [];
```
Comment (russellwheatley) [inline firestore-translate-text/functions/src/index.ts:144]: I think you can remove lines 144-157. Stringifying the input & checking for equality should be enough. Update line 159. i.e. `JSON.stringify(inputBefore) === JSON.stringify(inputAfter)`

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: firestore-translate-text/functions/src/index.ts (hunk 3 above)
Comment (dackers86) [reply firestore-translate-text/functions/src/index.ts:144]: I think you're right here. Will update and retest

**Comment by russellwheatley** [comment]
LGTM.

**Review by jhuleatt — APPROVED** [review]

**Comment by KeisukeNagakawa** [comment]
Hi! Is this functinoality available in extension?  I've tried like this but didn't work(this caused field name of "content,title" as the target field to translate. not "content" and "title" ). 
スクリーンショット 2022-03-07 15 50 47

**Pull request merged**